src/workerd/jsg/async-context.h

```diff
@@ -62,14 +62,15 @@ class AsyncContextFrame final: public Wrappable {
   // AsyncContextFrame::StorageKey is used to define a storage cell within the storage
   // context.
 public:
-  class StorageKey: public kj::Refcounted {
-    // An opaque key that identifies an async-local storage cell within the frame.
+  class StorageKey: public kj::AtomicRefcounted {
```

Comment:
AtomicRefcounted is quite expensive, let's see if we can avoid manipulating this outside of the isolate lock instead.